lib/cocoapods/bazel/target.rb

```diff
@@ -532,24 +532,28 @@ def expand_glob(glob, extensions: nil, expand_directories: false)
         elsif extensions && File.extname(glob).empty?
           glob = glob.chomp('**/*') # If we reach here and the glob ends with **/*, we need to avoid duplicating it (we do not want to end up with **/*/**/*)
           if File.basename(glob) == '*'
-            extensions.map { |ext| "#{glob}.#{ext}" }
+            extensions.map do |ext|
+              combined = "#{glob}.#{ext}"
+              combined if Dir.glob(File.join(@package_dir, combined)).any?
+            end.compact
           else
             extensions.map do |ext|
-              File.join(glob, '**', "*.#{ext}")
-            end
+              combined = File.join(glob, '**', "*.#{ext}")
+              combined if Dir.glob(File.join(@package_dir, combined)).any?
+            end.compact
           end
         elsif expand_directories
           if glob.end_with?('/**/*')
-            [glob]
+            Dir.glob(File.join(@package_dir, glob)).any? ? [glob] : []
```

Comment:
Can we extract this pattern to a small helper function since it's being used in diff places?

Reply:
Updated!